File: tree/0226_invert_binary_tree/0226_invert_binary_tree.go
package main

import "fmt"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func invertTree(root *TreeNode) *TreeNode {
	if root == nil || (root.Left == nil && root.Right == nil) {
		return root
	}

	root.Left, root.Right = invertTree(root.Right), invertTree(root.Left)

	return root
}

func main() {
	root := &TreeNode{1, &TreeNode{2, nil, nil}, &TreeNode{3, nil, nil}}
	fmt.Println(invertTree(root).Right.Val)
}

File: tree/0226_invert_binary_tree/0226_invert_binary_tree.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution(object):
    def invertTree(self, root):
        if root is not None:
            root.left , root.right = self.invertTree(root.right), self.invertTree(root.left)
        
        return root

root = TreeNode(1)
root.left = TreeNode(2)
root.right = TreeNode(3)
res = Solution().invertTree(root)
print(res.right.val)